warp10/src/main/java/io/warp10/script/functions/SNAPSHOT.java

```diff
@@ -411,6 +416,37 @@ public static void addElement(SNAPSHOT snapshot, StringBuilder sb, Object o, boo
         sb.append("' } ");
         sb.append(WarpScriptLib.RSAPRIVATE);
         sb.append(" ");
+      } else if (o instanceof PGPPublicKey) {
+        try {
+          addElement(sb, ((PGPPublicKey) o).getEncoded(false));
+        } catch (IOException ioe) {
+          throw new WarpScriptException("Error while serializing PGP public key.", ioe);
+        }
+        sb.append(WarpScriptLib.PGPPUBLIC);
+        sb.append(" ");
+        String keyid = "000000000000000" + Long.toHexString(((PGPPublicKey) o).getKeyID());
+        keyid = keyid.substring(keyid.length() - 16, keyid.length()).toUpperCase();
+        addElement(sb, keyid);
+        sb.append(WarpScriptLib.GET);
+        sb.append(" ");
+        addElement(sb, PGPPUBLIC.KEY_KEY);
+        sb.append(WarpScriptLib.GET);
+        sb.append(" ");
+      } else if (o instanceof PGPSecretKey) {
+        try {
+          addElement(sb, ((PGPSecretKey) o).getEncoded());
+        } catch (IOException ioe) {
+          throw new WarpScriptException("Error while serializing PGP secret key.", ioe);
+        }
+        sb.append(WarpScriptLib.PGPPRIVATE);
+        sb.append(" ");
+        String keyid = "000000000000000" + Long.toHexString(((PGPSecretKey) o).getKeyID());
+        keyid = keyid.substring(keyid.length() - 16, keyid.length()).toUpperCase();
+        addElement(sb, keyid);
+        sb.append(WarpScriptLib.GET);
+        sb.append(" ");
+        addElement(sb, PGPPUBLIC.KEY_KEY);
+        sb.append(WarpScriptLib.GET);
```

Comment:
missing a space afer GET  :  `'key'GET`

Reply:
The following example is still not working (even with the space added), there is a problem with the second key (PGPPRIVATE error decoding secret key. (unknown object in stream: 7))


```
'-----BEGIN PGP PRIVATE KEY BLOCK-----%0AComment: GPGTools - http://gpgtools.org%0A%0AlIYEYzVt3RYJKwYBBAHaRw8BAQdAB4Kf26riz5c9oMBpO2aO5IjKaA9CrA9jNMXW%0A3cdP1LT+BwMCWsdG+aMSBFLxznQ3yo/8iIExHvf40gNWxH2gQFMLJK2ulRjUHk70%0A1Pa6kiZ+Yh2BWe6O9ikA/Cx+xcyBNGOpv+PrcBn7kb34Qly5qWecYbQVdG90byA8%0AdG90b0B0b3RvLnRvdG8+iJYEExYKAD4WIQQ7vNsvm5T5/pxyXpwsnSKsc1Ol5QUC%0AYzVt3QIbAwUJA8JnAAULCQgHAwUVCgkICwUWAwIBAAIeBQIXgAAKCRAsnSKsc1Ol%0A5etnAQD9NOGZISnOrFohonGRu020t6aoiKAKXp23j2hkx3tmfgD+IPESn2VuOpCq%0AAO36EicRTvReAybMrspCwmPp/NTyYwCciwRjNW3dEgorBgEEAZdVAQUBAQdAOy86%0A9sdZChk+4lOdYwV/2Ow1YeMefp90/Td0XByqVQcDAQgH/gcDAlaFYuLs7s5m8elI%0AGfzv6kGjdoqgQ+6c1mtpTONkQBJQUlmPgnNme69YXKvGTcFCNEnG5NrYI0v9rK15%0AEM1WGwX/bMVWFHgRcl30+yb+CXSIfgQYFgoAJhYhBDu82y+blPn+nHJenCydIqxz%0AU6XlBQJjNW3dAhsMBQkDwmcAAAoJECydIqxzU6XlscoBAILeu70vHldvJDpZpw3O%0A5Ju1Wze0HXPrfhIDOFMO1d4jAQCs8GcxKsdCYzqWRtvZ27qaRqc1ekOEzP7ljurx%0AmDzrDg==%0A=vN0f%0A-----END PGP PRIVATE KEY BLOCK-----'
PGPPRIVATE

SNAPSHOT  EVAL 
```